frontend/src/views/BannerAnnouncement.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <div v-if="showBanner" :class="[`${bgColor}`]">
+    <div
+      class="mx-auto py-1 px-3 w-full flex flex-row items-center justify-center flex-wrap"
+    >
+      <div class="py-1 px-3 font-medium text-white flex flex-row items-center">
+        <p>{{ announcementText }}</p>
+      </div>
+      <div
+        v-if="announcementLink.length > 0"
+        class="item-center py-1 px-3 font-medium text-white truncate"
+      >
+        <a
+          :href="announcementLink"
+          target="_blank"
+          class="text-center underline"
+        >
+          <svg
```

Comment:
I've no idea what this icon is. If it's a public resource from heroicons, we may use `<heroicons:xxxx>` directly to make it easy and readable. If it is a special icon, we may extract it to a single SVG file in the assets folder to make it preview-able.